api/filters.go

```diff
@@ -36,8 +37,21 @@ func getSortFromString(str string) int {
 func CreateFilterFromQueryParam(d rules.Queryable, params url.Values) db.FindFilter {
 	filter := db.FindFilter{}
 
-	filter.Page = parseIntFromString(params.Get("page"))
-	filter.PerPage = parseIntFromString(params.Get("perPage"))
+	page := 1
+	if params.Get("page") != "" {
+		page = parseIntFromString(params.Get("page"))
+	}
+	filter.Page = page
+
+	perPage := viper.GetInt("db.perPage")
+
+	if params.Get("perPage") != "" {
+		perPage = parseIntFromString(params.Get("perPage"))
```

Comment:
If we have an invalid perPage, we will get a perPage=0 (non-ideal behavior).
We could refactor parseIntFromString to receive a default value in order to fix it

Reply:
good catch, thank you!